addons/mail/models/res_users.py

```diff
@@ -82,37 +82,38 @@ def create(self, vals_list):
         users = super(Users, self).create(vals_list)
         # Auto-subscribe to channels unless skip explicitly requested
         if not self.env.context.get('mail_channel_skip_auto_subscribe'):
-            self.env['mail.channel'].search([('group_ids', 'in', users.groups_id.ids)])._subscribe_users()
+            self.env['mail.channel'].search([('group_ids', 'in', users.groups_id.ids)])._subscribe_users_automatically()
         return users
 
     def write(self, vals):
         write_res = super(Users, self).write(vals)
         if 'active' in vals and not vals['active']:
-            self._unsubscribe_from_channels()
+            self._unsubscribe_from_non_public_channels()
         sel_groups = [vals[k] for k in vals if is_selection_groups(k) and vals[k]]
         if vals.get('groups_id'):
             # form: {'group_ids': [(3, 10), (3, 3), (4, 10), (4, 3)]} or {'group_ids': [(6, 0, [ids]}
             user_group_ids = [command[1] for command in vals['groups_id'] if command[0] == 4]
             user_group_ids += [id for command in vals['groups_id'] if command[0] == 6 for id in command[2]]
-            self.env['mail.channel'].search([('group_ids', 'in', user_group_ids)])._subscribe_users()
+            self.env['mail.channel'].search([('group_ids', 'in', user_group_ids)])._subscribe_users_automatically()
         elif sel_groups:
-            self.env['mail.channel'].search([('group_ids', 'in', sel_groups)])._subscribe_users()
+            self.env['mail.channel'].search([('group_ids', 'in', sel_groups)])._subscribe_users_automatically()
         return write_res
 
     def unlink(self):
-        self._unsubscribe_from_channels()
+        self._unsubscribe_from_non_public_channels()
         return super().unlink()
 
-    def _unsubscribe_from_channels(self):
+    def _unsubscribe_from_non_public_channels(self):
         """ This method un-subscribes users from private mail channels. Main purpose of this
             method is to prevent sending internal communication to archived / deleted users.
             We do not un-subscribes users from public channels because in most common cases,
             public channels are mailing list (e-mail based) and so users should always receive
             updates from public channels until they manually un-subscribe themselves.
         """
-        self.mapped('partner_id.channel_ids').filtered(lambda c: c.public != 'public').write({
-            'channel_partner_ids': [Command.unlink(pid) for pid in self.mapped('partner_id').ids]
-        })
+        self.env['mail.channel.partner'].sudo().search([
```

Comment:
There is some wild amount of "sudo" added in this commit but I'm guessing it's intended?

Reply:
Well I wanted to make it a completely technical model but then thought it could be done post-merge in another cleaning step. Some sudo are still there, especially in "technical" code indeed. In short that's intended even if not completely mandatory.

Reply:
I'm still not sure this should be done. I feel like all users should have access to their own channel partner.
But then it's more of a generic matter of do we want to keep access rules, or do we want to block everything and only use sudo when necessary.

Reply:
@seb-odoo reading is ok. Modifying / Creating / Unlinking should not be available outside of some specific actions / flows. Purpose is to give as less space to spoofing as possible. Note that unlinking its own membership could be considered as ok. Anyway worth another task :D .